### DISCUSSION TITLE
Changing Filenames to not include a colon

### DISCUSSION THREAD

**Discussion by gniknalu** [roflcoopter/viseron#181] [Q&A] [answered]
[Re]installed Viseron because I noticed it now uses YOLO4. **So far it's AWESOME!**

But the filenames are unusable.
In the logs, for file creation, it shows:
`[2021-05-25 16:38:54] [lib.recorder.frontwest  ] [DEBUG   ] - Concatenation command: ['ffmpeg', '-hide_banner', '-loglevel', 'error', '-y', '-protocol_whitelist', 'file,pipe', '-f', 'concat', '-safe', '0', '-i', '-', '-c:v', 'copy', '-movflags', '+faststart', '/recordings/2021-05-25/FrontWest/16:38:41.mp4']`

But, because of the colons, the filename in the samba share appears as:
159ZX1~2.MP4

In samba they can be mapped to other characters but, even then, they're not opening correctly.

**My Question:** Is there a way to change the filename so they don't include colon? (Can't find a setting in the documentation as of 2021.05.25) Something like %Y-%m-%d %f

Thank you

**Comment by roflcoopter** [comment]
Hi! 

There is no way to do this right now, but i (think) it will be an easy fix, i will look into it asap.
Could you open an issue so i can track it properly?

Also on the YOLOv4, are you running the amd64-image? Next version will once again default to YOLOv3 due to perfomance reasons, but YOLOv4 is included in the image so you will be able to switch to that using this config:
```yaml
detector:
  type: darknet
  model_path: /detectors/models/darknet/yolov4.weights
  model_config: /detectors/models/darknet/yolov4.cfg
```

**Reply by gniknalu** [reply]
Thank you for responding! Thank you for looking into the fix. I will open the issue for tracking.

Yes - YOLO4, in my installation - using the amd64 image - has a very low load on my Nvidia GPU as compared to the tensorflow and deepstack images I've tried from other apps.

I'm happy to see there will be a way to keep using YOLO4! 

**Thank you for your good work!**

**Comment by gniknalu** [comment]
In recorder, the "filename_pattern" - in the dev version - only applies to the video but not the detected image. Is there a setting for that? 

**Thank you!**

**Reply by roflcoopter** [reply]
The thumbnail also has that config option, if you view the README and switch to the dev branch you can find it there. Is that what you meant?

**Reply by gniknalu** [reply]
> The thumbnail also has that config option, if you view the README and switch to the dev branch you can find it there. Is that what you meant?

Figured out how to do that. My apologies. **AND my Thanks!**

**Reply by gniknalu** [reply]
> The thumbnail also has that config option, if you view the README and switch to the dev branch you can find it there. Is that what you meant?

I've tried all kinds of combinations to get "thumbnail" to work. Can you please, when you get an opportunity, post an example that includes thumbnail options? I tried it under "recorder" and I've tried it stand-alone. In both instances I've varied my YAML to see if it was a formatting issue but neither worked with the dev branch.

**THANK YOU**

**Reply by roflcoopter** [reply]
I'm not at home right now so this is untested but it should be like this: 
```yaml
recorder:
  thumbnail:
    save_to_disk: true
    filename_pattern: <your pattern here>
```

If it doesn't work, is there any error message in the log?

**Reply by gniknalu** [reply]
I didn't try **THAT** combination. I was using this:
```
recorder:
  thumbnail:
    - filename_pattern:
```

and this:

```
recorder:
  - thumbnail:
       filename_pattern: 
```

But what you describe - of course - works!

Thanks again!

**Reply by gniknalu** [reply]
> I'm not at home right now so this is untested but it should be like this:
> 
> ```yaml
> recorder:
>   thumbnail:
>     save_to_disk: true
>     filename_pattern: <your pattern here>
> ```
> 
> If it doesn't work, is there any error message in the log?

How do I mark this as "Answered"?

**Reply by roflcoopter** [reply]
seems like you cant do that in a thread, need to post a new answer, ill fix it

**Comment by roflcoopter** [comment, marked as answer]
```yaml
recorder:
  thumbnail:
    save_to_disk: true
    filename_pattern: <your pattern here>
```
